src/Utils.ps1

```diff
@@ -216,6 +216,123 @@ function Add-PoshGitToProfile {
+        $AllUsers,
+
+        [Parameter(ValueFromRemainingArguments)]
+        [psobject[]]
+        $TestParams
+    )
+
+    if ($AllUsers -and !(Test-Administrator)) {
+        throw 'Removing posh-git from an AllUsers profile requires an elevated host.'
+    }
+
+    $underTest = $false
+
+    $profileName = $(if ($AllUsers) { 'AllUsers' } else { 'CurrentUser' }) `
+                 + $(if ($AllHosts) { 'AllHosts' } else { 'CurrentHost' })
+    Write-Verbose "`$profileName = '$profileName'"
+
+    $profilePath = $PROFILE.$profileName
+    Write-Verbose "`$profilePath = '$profilePath'"
+
+    # Under test, we override some variables using $args as a backdoor.
+    if (($TestParams.Count -gt 0) -and ($TestParams[0] -is [string])) {
+        $profilePath = [string]$TestParams[0]
+        $underTest = $true
+        if ($TestParams.Count -gt 1) {
+            $ModuleBasePath = [string]$TestParams[1]
+        }
+    }
+
+    if (!$profilePath) { $profilePath = $PROFILE }
+
+    if (!$profilePath) {
+        Write-Warning "Skipping add of posh-git import to profile; no profile found."
+        Write-Verbose "`$PROFILE              = '$PROFILE'"
+        Write-Verbose "CurrentUserCurrentHost = '$($PROFILE.CurrentUserCurrentHost)'"
+        Write-Verbose "CurrentUserAllHosts    = '$($PROFILE.CurrentUserAllHosts)'"
+        Write-Verbose "AllUsersCurrentHost    = '$($PROFILE.AllUsersCurrentHost)'"
+        Write-Verbose "AllUsersAllHosts       = '$($PROFILE.AllUsersAllHosts)'"
+        return
+    }
+
+    if (Test-Path -LiteralPath $profilePath) {
+        # If the profile script exists and is signed, then we should not modify it
+        if (!(Get-Command Get-AuthenticodeSignature -ErrorAction SilentlyContinue))
+        {
+            Write-Verbose "Platform doesn't support script signing, skipping test for signed profile."
+        }
+        else {
+            $sig = Get-AuthenticodeSignature $profilePath
+            if ($null -ne $sig.SignerCertificate) {
+                Write-Warning "Skipping add of posh-git import to profile; '$profilePath' appears to be signed."
+                Write-Warning "Add the command 'Import-Module posh-git' to your profile and resign it."
+                return
+            }
+        }
+
+        $oldProfile = @(Get-Content $profilePath)
+
+        . $currentVersionPath\src\Utils.ps1
+        $oldProfileEncoding = Get-FileEncoding $profilePath
+
+        $newProfile = @()
+        foreach($line in $oldProfile) {
+            if ($line -like '*PoshGitPrompt*') { continue; }
+            if ($line -like '*Load posh-git example profile*') { continue; }
+
+            if($line -like '. *posh-git*profile.example.ps1*') {
+                continue;
+            }
+            if($line -like 'Import-Module *\posh-git.psd1*') {
+                continue;
+            }
+            $newProfile += $line
+        }
+        Set-Content -path $profilePath -value $newProfile -Force -Encoding $oldProfileEncoding
```

Comment:
Noting this is pulled from: https://github.com/dahlbyk/posh-git/blob/dc2b6367324a9db79db55186375462a454f29c2e/chocolatey/tools/chocolateyUninstall.ps1#L7-L25

Will be nice to use this function there.